cmd/srcd/cmd/root.go

```diff
@@ -88,37 +88,89 @@ func initConfig() {
 
 var logMsgRegex = regexp.MustCompile(`.*msg="(.+)"`)
 
-func logAfterTimeout(header string) chan struct{} {
-	logs, err := daemon.GetLogs()
-	if err != nil {
-		logrus.Errorf("could not get logs from server container: %v", err)
+func logAfterTimeout(msg string, timeout time.Duration) func() {
+	d := &Defered{
+		Timeout: timeout,
+		Msg:     msg,
 	}
 
-	started := make(chan struct{})
-	go func() {
-		defer logs.Close()
+	return d.Print()
+}
+
+func logAfterTimeoutWithSpinner(msg string, timeout time.Duration, spinnerInterval time.Duration) func() {
```

Comment:
any reason to for `spinnerInterval` argument here?

Reply:
Just to give the caller the flexibility to change it.

Reply:
up to you. I just prefer to have constructors only with reasonable defaults. The rest can be configured directly by using the struct instead.
Though not hard feelings here, keep it if you like :)